amundsen_application/static/js/components/SearchPage/index.tsx

```diff
@@ -59,67 +63,106 @@ class SearchPage extends React.Component<SearchPageProps, SearchPageState> {
-      results: [{
-        type: ResourceType.user,
-        active: true,
-        birthday: '10-10-1990',
-        department: "Department",
-        email: "mail@email.com",
-        first_name: "Ash",
-        github_username: "Github User",
-        id: 12345,
-        last_name: "Ketchum",
-        manager_email: "manager@company.com",
-        name: "Ash Ketchum",
-        offboarded: false,
-        office: "city",
-        role: "Pokemon Trainer",
-        start_date: "10-10-2014",
-        team_name: "Team Red",
-        title: "Pikachu",
-      },{
-        type: ResourceType.user,
-        active: false,
-        birthday: '10-10-1990',
-        department: "Department",
-        email: "mail@email.com",
-        first_name: "Gary",
-        github_username: "Github User",
-        id: 12345,
-        last_name: "Oak",
-        manager_email: "manager@company.com",
-        name: "Gary Oak",
-        offboarded: false,
-        office: "city",
-        role: "Pokemon Trainer",
-        start_date: "10-10-2014",
-        team_name: "Team Blue",
-        title: "Eevee",
-      }],
-      total_results: 1,
+      results: [],
+      total_results: 0,
     }
   };
 
   constructor(props) {
     super(props);
 
-    this.handlePageChange = this.handlePageChange.bind(this);
-    this.updateQueryString = this.updateQueryString.bind(this);
+    this.state = {
+      selectedTab: ResourceType.table,
+    };
   }
 
   componentDidMount() {
     this.props.getPopularTables();
 
     const params = qs.parse(window.location.search);
-    const searchTerm = params['searchTerm'];
-    const pageIndex = params['pageIndex'];
+    const { searchTerm, pageIndex, selectedTab} = params;
+
+    this.updateSelectedTab(selectedTab);
     if (searchTerm && searchTerm.length > 0) {
       const index = pageIndex || '0';
-      this.props.executeSearch(searchTerm, index);
+      this.props.searchAll(searchTerm, this.getSearchOptions(index));
     }
   }
 
+  updateSelectedTab = (newTab) => {
+    switch(newTab) {
+      case ResourceType.table:
+      case ResourceType.user:
+        this.setState({ selectedTab: newTab });
+        break;
+      case ResourceType.dashboard:
+        break;
+    }
+  };
+
+  getSearchOptions = (pageIndex) => {
+    return {
+      dashboardIndex: (this.state.selectedTab === ResourceType.dashboard) ? pageIndex : 0,
+      userIndex: (this.state.selectedTab === ResourceType.user) ? pageIndex : 0,
+      tableIndex: (this.state.selectedTab === ResourceType.table) ? pageIndex : 0,
+    };
+  };
+
+  getPageIndex = (tab) => {
+    switch(tab) {
+      case ResourceType.table:
+        return this.props.tables.page_index;
+      case ResourceType.user:
+        return this.props.users.page_index;
+      case ResourceType.dashboard:
+        return this.props.dashboards.page_index;
+    }
+    return 0;
+  };
+
+  onSearchBarSubmit = (searchTerm: string) => {
+    this.props.searchAll(searchTerm, this.getSearchOptions(0));
```

Comment:
nit: if pageIndex is `0` we could just return some default options, or perhaps the default search options should be baked into `this.props.searchAll()`. That way we don't have to run through the logic in `this.getSearchOptions` just to get the known output `{ dashboardIndex: 0, userIndex: 0, tableIndex: 0}`

Reply:
I made each field inside the `options` object optional, but I did not make the 'options' object itself optional. Will update.